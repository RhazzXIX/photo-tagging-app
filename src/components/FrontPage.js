import { Link } from "react-router-dom";

export default function FrontPage (props) {
  const { img } = props
  return(
    <main id="FrontPage">
      <h1>Seek & Click</h1>
      <button>Play!</button>
      <img src={img.src} alt={img.name} />
        <p>
          -----&gt; &nbsp;
      <Link to='kids'>
        Kids ver.
      </Link>
        </p>
    </main>
  )
}