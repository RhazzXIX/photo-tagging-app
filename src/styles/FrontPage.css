main#FrontPage {
  flex: auto;
  display: grid;
  grid-template-rows: 1fr 1fr;
  grid-template-columns: 1fr 1fr;
  justify-items: center;
  align-items: center;
  background-color: #22107f;
}

div#hContainer {
  width: 320px;
  height: 200px;
  background-color: rgba(255, 255, 255, 0.8);
  border-radius: 20px;
  color: #22107f;
  display: flex;
  flex-direction: column;
  justify-content: center;
  gap: 6px;
  padding: 0 18px;
  box-shadow: 0 0 8px rgb(197, 197, 197);
}

div#hContainer h1 {
  font-size: 3rem;
  font-family: Georgia, "Times New Roman", Times, serif;
  letter-spacing: 2px;
}

div#hContainer h1:nth-of-type(2) {
  align-self: center;
}

div#hContainer h1:nth-of-type(3) {
  align-self: flex-end;
}

main#FrontPage > video {
  grid-column: 2 / 3;
  grid-row: 1 / -1;
  border-radius: 8px;
  padding: 8px;
  background-color: rgba(255, 255, 255, 0.7);
  box-shadow: 0 0 8px rgb(197, 197, 197);
}

main#FrontPage > p,
a.version {
  grid-column: 2 / 3;
  grid-row: 2 / 3;
  color: yellow;
}

main#FrontPage > a button {
  font-size: 3rem;
  font-weight: 700;
  padding: 4px 50px;
  border-radius: 50px;
  border: 2px solid black;
  background-color: purple;
  color: yellow;
  font-family: Georgia, "Times New Roman", Times, serif;
  letter-spacing: 2px;
  cursor: pointer;
}

main#FrontPage > a button:hover {
  box-shadow: 0 0 10px white;
}

main#FrontPage > a button:active {
  background-color: aliceblue;
  color: #22107f;
}
