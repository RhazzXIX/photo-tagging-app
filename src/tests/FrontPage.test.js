import '@testing-library/jest-dom'
import { screen, render } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { BrowserRouter } from 'react-router-dom'
import FrontPage from '../components/FrontPage'

describe('FrontPage component' , () => {
  it('Renders on screen', () => {
    const { container } = render(
    <BrowserRouter>
      <FrontPage img={{}}/>
    </BrowserRouter>
    )
    expect(container).toMatchSnapshot()
  })
})