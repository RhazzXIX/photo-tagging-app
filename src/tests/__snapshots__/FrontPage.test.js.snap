// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FrontPage component Renders on screen 1`] = `
<div>
  <main
    id="FrontPage"
  >
    <h1>
      Seek & Click
    </h1>
    <button>
      Play!
    </button>
    <img />
    <p>
      -----\`&gt;\`
      <a
        href="/kids"
      >
        Kids ver.
      </a>
    </p>
  </main>
</div>
`;
